### PR TITLE
Supertokens integration

### PR DESCRIPTION
**Pull request by nkshah2**

### CODE REVIEW

File: packages/auth/src/authClients/supertokens.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    client: undefined,
+    login: async () => client.authRecipe.redirectToAuth('signin'),
+
+    signup: async () => client.authRecipe.redirectToAuth('signup'),
+
+    logout: async () => client.sessions.signOut(),
+
+    getToken: async (): Promise<string | null> => {
+      if (await client.sessions.doesSessionExist()) {
+        let jwtPropertyName = client.jwtPropertyName !== undefined && client.jwtPropertyName !== null ? client.jwtPropertyName : "jwt";
```
Comment (rishabhpoddar) [inline packages/auth/src/authClients/supertokens.ts:28]: We don't need to make the user pass `jwtPropertyName` cause we can know if from `_jwtPName`

File: packages/auth/src/authClients/supertokens.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        return {
+          userId: await client.sessions.getUserId(),
+          accessTokenPayload: await client.sessions.getAccessTokenPayloadSecurely(),
+        }
+      } else {
+        return null;
+      }
+    },
+    restoreAuthState: async () => {
+      return client.sessions.attemptRefreshingSession();
```
Comment (rishabhpoddar) [inline packages/auth/src/authClients/supertokens.ts:48]: You don't need to impl this function at all (i think)
Test by removing access token and reloading the page..

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: packages/auth/src/authClients/supertokens.ts (hunk 1 above)
Comment (nkshah2) [reply packages/auth/src/authClients/supertokens.ts]: Done

File: packages/auth/src/authClients/supertokens.ts (hunk 2 above)
Comment (nkshah2) [reply packages/auth/src/authClients/supertokens.ts]: Done

**Review by rishabhpoddar — APPROVED** [review]

File: packages/auth/src/authClients/supertokens.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+export interface SuperTokensUser {
+  userId: string;
+  accessTokenPayload: any;
+}
+
+export type SuperTokens = AuthClient;
+
+export const supertokens = (client: {
+  authRecipe: any,
+  sessions: typeof Sessions,
```
Comment (rishabhpoddar) [inline packages/auth/src/authClients/supertokens.ts:14]: Wd don't need this as an input cause the `Sessions` being imported is a singleton anyway. So below, we can just do `Sessions.signOut()`, or `Sessions.doesSessionExist()` etc..

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**